### PR TITLE
[no merge] comments on limiter and encoding

### PR DESCRIPTION
**Pull request by longbowlu**

### CODE REVIEW

File: contracts/BridgeLimiter.sol [hunk 1]
```diff
@@ -55,6 +55,9 @@ contract BridgeLimiter is
         return windowAmount + USDAmount > totalLimit;
     }
 
+    // 1. We should look at hourlyTransferAmount[currentHour] because that's the most likely
+    // time window a large amount of fund goes out if bridge is hacked
+    // 2. rename to `calculateLast24hAmount` ?
```
Comment (Bridgerz) [inline contracts/BridgeLimiter.sol:60]: I named it this with the intention to enable the window length to be initialized. 

I do think calculateWindowAmount is better than calculateLast24hAmount.... nothing against your naming conventions 😋.

File: contracts/BridgeLimiter.sol [hunk 2]
```diff
@@ -55,6 +55,9 @@ contract BridgeLimiter is
         return windowAmount + USDAmount > totalLimit;
     }
 
+    // 1. We should look at hourlyTransferAmount[currentHour] because that's the most likely
```
Comment (Bridgerz) [inline contracts/BridgeLimiter.sol:58]: I'm not sure what you mean by this. the first hour we look at is hourlyTransferAmount[currentHour - 0]

File: contracts/BridgeLimiter.sol [hunk 3]
```diff
@@ -64,6 +67,7 @@ contract BridgeLimiter is
         return total;
     }
 
+    // FIXME
```
Comment (Bridgerz) [inline contracts/BridgeLimiter.sol:70]: whats wrong here?

File: contracts/BridgeLimiter.sol [hunk 4]
```diff
@@ -75,24 +79,30 @@ contract BridgeLimiter is
 
     /* ========== EXTERNAL FUNCTIONS ========== */
 
+    // there are duplicated calls to `calculateAmountInUSD`, one in `willAmountExceedLimit`
+    // and one in in line 100. Can we modify `willAmountExceedLimit` to let it take the USD amount instead?
```
Comment (Bridgerz) [inline contracts/BridgeLimiter.sol:83]: not a bad idea. will look into this.

File: contracts/BridgeLimiter.sol [hunk 5]
```diff
@@ -75,24 +79,30 @@ contract BridgeLimiter is
...
     function updateBridgeTransfers(uint8 tokenId, uint256 amount) external override onlyOwner {
         require(amount > 0, "BridgeLimiter: amount must be greater than 0");
         require(
             !willAmountExceedLimit(tokenId, amount),
             "BridgeLimiter: amount exceeds rolling window limit"
         );
 
+        // We have this in a few places as well
```
Comment (Bridgerz) [inline contracts/BridgeLimiter.sol:91]: not a bad idea to add a "currentHour" function.

File: contracts/BridgeLimiter.sol [hunk 6]
```diff
@@ -75,24 +79,30 @@ contract BridgeLimiter is
...
         uint32 currentHour = uint32(block.timestamp / 1 hours);
 
         // garbage collect most recently expired hour if window is moving
+        // why do we need `hourlyTransferAmount[currentHour] == 0`?
```
Comment (Bridgerz) [inline contracts/BridgeLimiter.sol:95]: It was there to check if the current hour has already accrued transfers (confirming the window has moved for the first time and garbage collection can take place) but I guess the "oldestHourTimestamp" (which needs a new name) does that verification too... nice catch.

File: contracts/BridgeLimiter.sol [hunk 7]
```diff
@@ -75,24 +79,30 @@ contract BridgeLimiter is
...
         if (hourlyTransferAmount[currentHour] == 0 && oldestHourTimestamp < currentHour - 24) {
+            // are the parameters correct?
```
Comment (Bridgerz) [inline contracts/BridgeLimiter.sol:97]: we want to garbage collect the most recent hour to expire (the 25th hour away from the current hour). The first parameter is the starting hour to try and garbagecollect, the second hour is the last hour to try and garbagecollect.

File: contracts/BridgeLimiter.sol [hunk 8]
```diff
@@ -75,24 +79,30 @@ contract BridgeLimiter is
...
             garbageCollectHourlyTransferAmount(currentHour - 25, currentHour - 25);
         }
 
         // update hourly transfers
         hourlyTransferAmount[currentHour] += calculateAmountInUSD(tokenId, amount);
     }
 
+    // I don't understand this function, primarily because I don't understand the args passed in a few lines above
```
Comment (Bridgerz) [inline contracts/BridgeLimiter.sol:105]: name needs to be more explicit (and some documentation wouldn't hurt lol). this function exists to enable the removal of expired hourly transfers (transfers older than 24 hours). the parameters resemble a range of hours to iterate through to try and remove from storage.

File: contracts/SuiBridge.sol [hunk 9]
```diff
@@ -269,6 +269,8 @@ contract SuiBridge is
             vault.transferERC20(tokenAddress, targetAddress, amount);
         }
 
+        // Do we want to move the limit checks earlier, e.g. even for signature verification?
+        // in the event of exhaustion of transfer allowance, the user can pay fewer gas fees
```
Comment (Bridgerz) [inline contracts/SuiBridge.sol:273]: I wouldn't do this in the signature verification, but we could add a "limit check" modifier to the "transferTokensWithSignatures" function that would accomplish the same thing but keep the separation of responsibility a bit cleaner. If we added a "willUSDAmountExceedLimit" (like mentioned above) we would also need to add a new "willTokenAmountExceedLimit" so the USD logic stays in the Limiter contract.

File: contracts/utils/BridgeMessage.sol [hunk 10]
```diff
@@ -93,6 +93,7 @@ library BridgeMessage {
         pure
         returns (address, bytes memory)
     {
+        // should it be abi.decodePacked()
```
Comment (Bridgerz) [inline contracts/utils/BridgeMessage.sol:96]: no. Will have George start on these.

File: contracts/utils/BridgeMessage.sol [hunk 11]
```diff
@@ -186,6 +187,7 @@ library BridgeMessage {
 
     // TODO: add unit test
     function decodeEmergencyOpPayload(bytes memory payload) internal pure returns (bool) {
+        // should it be abi.decodePacked()
```
Comment (Bridgerz) [inline contracts/utils/BridgeMessage.sol:190]: dido

File: contracts/utils/BridgeMessage.sol [hunk 12]
```diff
@@ -204,6 +206,7 @@ library BridgeMessage {
         pure
         returns (bool, address[] memory)
     {
+        // should it be abi.decodePacked()
```
Comment (Bridgerz) [inline contracts/utils/BridgeMessage.sol:209]: dido

File: contracts/utils/BridgeMessage.sol [hunk 13]
```diff
@@ -216,11 +219,15 @@ library BridgeMessage {
         pure
         returns (uint8, uint256)
     {
+    // 1. could price smaller uint?
+    // 2. should it be abi.decodePacked()
+    // 3. do we want uint256 or whatever type to be big endian?
```
Comment (Bridgerz) [inline contracts/utils/BridgeMessage.sol:224]: 1. possibly uint64?
2. no
3. big endian from here on out is preferable.

File: contracts/BridgeLimiter.sol (hunk 2 above)
Comment (longbowlu) [reply contracts/BridgeLimiter.sol:58]: I don't understand what i meant either. Perhaps I misread the code?

File: contracts/BridgeLimiter.sol (hunk 1 above)
Comment (longbowlu) [reply contracts/BridgeLimiter.sol:60]: fair

File: contracts/SuiBridge.sol (hunk 9 above)
Comment (longbowlu) [reply contracts/SuiBridge.sol:273]: sounds good to me. Do you think the gas saving is worth this change?

File: contracts/utils/BridgeMessage.sol (hunk 13 above)
Comment (longbowlu) [reply contracts/utils/BridgeMessage.sol:224]: passing the order of big endian to @patrickkuo lol

File: contracts/BridgeLimiter.sol (hunk 8 above)
Comment (longbowlu) [reply contracts/BridgeLimiter.sol:105]: extract 
```
        for (uint32 i = startHour; i <= endHour; i++) {
            if (hourlyTransferAmount[i] > 0) delete hourlyTransferAmount[i];
        }
        // update oldest hour if current oldest hour was garbage collected
        if (startHour == oldestHourTimestamp) {
            oldestHourTimestamp = endHour + 1;
        }
```
to its own function

**Pull request closed without merging**